File: exercism/elixir/allergies/lib/allergies.ex
defmodule Allergies do
  @allergy_map %{
    "eggs" => 1,
    "peanuts" => 2,
    "shellfish" => 4,
    "strawberries" => 8,
    "tomatoes" => 16,
    "chocolate" => 32,
    "pollen" => 64,
    "cats" => 128
  }

  @doc """
  List the allergies for which the corresponding flag bit is true.
  """
  @spec list(non_neg_integer) :: [String.t()]
  def list(flags) do
    @allergy_map
    |> Enum.filter(fn {k, _} -> allergic_to?(flags, k) end)
    |> Enum.map(fn {k, _} -> k end)
  end

  @doc """
  Returns whether the corresponding flag bit in 'flags' is set for the item.
  """
  @spec allergic_to?(non_neg_integer, String.t()) :: boolean
  def allergic_to?(flags, item), do: @allergy_map |> Map.get(item, 0) |> Bitwise.&&&(flags) !== 0
end

File: exercism/cpp/allergies/allergies.cpp
#include "allergies.h"

std::map<std::string, unsigned int> 
    allergies::allergy_test::allergies = {
        {"eggs", 0x01},
        {"peanuts", 0x02},
        {"shellfish", 0x04},
        {"strawberries", 0x08},
        {"tomatoes", 0x10},
        {"chocolate", 0x20},
        {"pollen", 0x40}, 
        {"cats", 0x80}};

namespace allergies {

bool allergy_test::is_allergic_to(std::string item) {
    return allergies[item] & score;
}

std::unordered_set<std::string> allergy_test::get_allergies() {
    std::unordered_set<std::string> set;
    for (const auto& kv: allergies)
        if (is_allergic_to(kv.first))
            set.emplace(kv.first);
    return set;
}

}
